### DISCUSSION TITLE
Assign multiple labels to AutoscalingRunnerSet

### DISCUSSION THREAD

**Discussion by dm3ch** [actions/actions-runner-controller#2831] [Questions]
Hi,
I'm trying to migrate from RunnerDeployment to AutoscalingRunnerSet .
On AutoscalingRunnerSet I haven't found how it's possible to manage runner labels. I would like to make my label standart: `[ self-hosted, ARM64, ...]`
Is it even possible?

Issue about the same problem - https://github.com/actions/actions-runner-controller/issues/2445#issuecomment-1692559904

**Comment by shakefu** [comment]
I have the same question - and they seem to be missing the default "self-hosted" label.

**Comment by shakefu** [comment]
From what I can tell here, it only supports 1 label, the `metadata.name` of the `AutoscalingRunnerSet`... https://github.com/actions/actions-runner-controller/blob/215b245881d944d45bf3d7fab3cb752cee0181ef/controllers/actions.github.com/autoscalingrunnerset_controller.go#L209

**Comment by shakefu** [comment]
The RunnerScaleSet type does in theory support multiple labels... https://github.com/actions/actions-runner-controller/blob/master/github/actions/types.go#L94C71-L94C71

**Reply by dm3ch** [reply]
I tried to patch the code https://github.com/actions/actions-runner-controller/blob/64cedff2b42790afc00f7ede5a9df897a1afb1ac/controllers/actions.github.com/autoscalingrunnerset_controller.go#L347-L352 to add one more static label just for test, but after that change the runner just started failing to register.

I don't know maybe I done it wrong or maybe API doesn't support that (runnersets uses a special separate api as far as I understood)
